daemon/container_operations_unix.go

```diff
@@ -63,7 +63,11 @@ func (daemon *Daemon) getIpcContainer(container *container.Container) (*containe
 	if err != nil {
 		return nil, errors.Wrapf(err, "cannot join IPC of a non running container: %s", container.ID)
 	}
-	return container, daemon.checkContainer(container, containerIsRunning, containerIsNotRestarting)
+	// Check that "donor" container ipc is shareable
+	if st, e := os.Stat(container.ShmPath); e == nil && st.IsDir() {
```

Comment:
os.Stat() can return so many errors, but in most cases it will be ENOENT, to which we should say "container IPC is not shareable". All other errors, well, fall under "unexpected error/this should not happen".

I changed the code based on that.